pkg/oci/oci.go

```diff
@@ -813,6 +818,32 @@ func ensureImage(ctx context.Context, imageStore oras.Target, image string, imgO
 		}
 	}
 
+	if len(imgOpts.AllowedDigests) > 0 {
+		targetImage, err := normalizeImageName(image)
+		if err != nil {
+			return fmt.Errorf("normalizing image: %w", err)
+		}
+
+		desc, err := imageStore.Resolve(ctx, targetImage.String())
+		if err != nil {
+			return fmt.Errorf("resolving image %q on local registry: %w", targetImage.String(), err)
+		}
+
+		digestString := desc.Digest.String()
+		found := false
+		for _, digest := range imgOpts.AllowedDigests {
+			if digestString == digest {
+				found = true
+
+				break
+			}
+		}
+
+		if !found {
+			return fmt.Errorf("image digest not allowed: %q not in %q", digestString, strings.Join(imgOpts.AllowedDigests, ", "))
```

Comment:
The `%q` with `strings.Join(imgOpts.AllowedDigests, ", ")` can potentially become VERY long (especially when those options come from a config file later on). I think just saying "image digest %q not found in allowed digests list" (or sth. like that) is enough.